### PR TITLE
ci: switch from hosted to local DB

### LINKED ISSUE #146: reduce number of events on a PR
@shaypal5 would it be possible to reduce the number of events in the PR? It is super long, and GH hides all intermediate comments, so the PR is much harder to follow... :confused: 
image

### PR DESCRIPTION
**Pull request by Borda**
resolves #146

### CODE REVIEW

**Comment by shaypal5** [comment]
Tell me if you need me to set up any secrets or environment variables in out GitHub Actions environments.

**Comment by Borda** [comment]
> Tell me if you need me to set up any secrets or environment variables in out GitHub Actions environments.

It shall not be needed; otherwise, it won't be changed, right?
seems we would need to run MongoDB with docker so it can run on all OS so far the two actions I tried were limited to  Linux only :/

**Comment by shaypal5** [comment]
> > Tell me if you need me to set up any secrets or environment variables in out GitHub Actions environments.
> 
> It shall not be needed; otherwise, it won't be changed, right? seems we would need to run MongoDB with docker so it can run on all OS so far the two actions I tried were limited to Linux only :/

Right you are! Silly me...

Well, let see if you can work the kinks out...

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -17,14 +17,19 @@ concurrency:
...
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         os: [ubuntu-latest, macOS-latest, windows-latest]
         backend: ["local", "db"]
+    env:
+      CACHIER_TEST_HOST: "localhost"
+      CACHIER_TEST_PORT: "27017"
+      #CACHIER_TEST_DB: "dummy_db"
+      #CACHIER_TEST_USERNAME: "myuser"
+      #CACHIER_TEST_PASSWORD: "yourpassword"
+      CACHIER_TEST_VS_LIVE_MONGO: "true"
```
Comment (Borda) [inline .github/workflows/test.yml:32]: not sure what this one shall do? @shaypal5

**Comment by Borda** [comment]
> Well, let see if you can work the kinks out...

@shaypal5 mind check t out, had to drop testing for DB on macOS as GH runners does not have reliable support for dockers yet and tested actions to add docker did not work well

**Review by shaypal5 — APPROVED** [review]
Amazing!

File: .github/workflows/test.yml (hunk 1 above)
Comment (shaypal5) [reply .github/workflows/test.yml:35]: I'm not sure myself. Maybe change it to "CACHIER_TEST_VS_DOCKERIZED_MONGO". And we can bring back this one if we ever want to test against a live MongoDB again.

**Review by shaypal5 — APPROVED** [review]
Great!

**Review by shaypal5 — CHANGES_REQUESTED** [review]
Just rename CACHIER_TEST_VS_LIVE_MONGO like I've suggested.

**Comment by Borda** [comment]
> Just rename CACHIER_TEST_VS_LIVE_MONGO like I've suggested.

ok, added it back such it works as a switch between local/test and live DB

**Comment by Borda** [comment]
@shaypal5 figured out the docker on Mac o it is a full matrix :flamingo:

**Review by shaypal5 — APPROVED** [review]

**Comment by shaypal5** [comment]
@Borda 
This cannot be rebased automatically from within GitHub, due to conflicts.
Please rebase manually and I'll squash and merge it.

**Pull request merged**